code/modules/projectiles/projectile/change.dm

```diff
@@ -9,91 +9,56 @@
 /obj/item/projectile/change/on_hit(var/atom/change)
 	wabbajack(change)
 
+/obj/item/projectile/change/proc/get_random_transformation_options(var/mob/M)
+	. = list()
+	if(!isrobot(M))
+		. += "robot"
+	for(var/t in get_all_species())
+		. += t
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		. -= H.species.name
+
+/obj/item/projectile/change/proc/apply_transformation(var/mob/M, var/choice)
+
+	if(choice == "robot")
+		var/mob/living/silicon/robot/R = new(get_turf(M))
+		R.gender = M.gender
+		R.job = "Robot"
+		R.mmi = new /obj/item/mmi(R)
+		R.mmi.transfer_identity(M)
+		return R
+
+	if(get_species_by_key(choice))
+		var/mob/living/carbon/human/H = M
+		if(!istype(H))
```

Comment:
shouldn't this also transfer player or something? Looks like it just creates a human on target's turf and calls it a day.

Reply:
They get mind transferred after apply_transformation is called.